tests/reader.rs

```diff
@@ -0,0 +1,75 @@
+use osmpbfreader;
+use std::fs::File;
+
+const PBF: &str = "tests/clipperton.osm.pbf";
+
+#[test]
+fn read_pbf() {
+    let file = File::open(PBF).unwrap();
+    let mut pbf = osmpbfreader::OsmPbfReader::new(file);
+
+    let mut num_nodes = 0;
+    let mut num_ways = 0;
+    let mut num_relations = 0;
+
+    for obj in pbf.iter() {
+        let obj = obj.unwrap();
+        match obj {
+            osmpbfreader::OsmObj::Node(_) => num_nodes += 1,
+            osmpbfreader::OsmObj::Way(_) => num_ways += 1,
+            osmpbfreader::OsmObj::Relation(_) => num_relations += 1,
+        }
+    }
+    assert_eq!(num_nodes, 3097);
+    assert_eq!(num_ways, 151);
+    assert_eq!(num_relations, 17);
+}
+
+#[test]
+fn read_pbf_par_iter() {
+    let file = File::open(PBF).unwrap();
+    let mut pbf = osmpbfreader::OsmPbfReader::new(file);
+
+    let mut num_nodes = 0;
+    let mut num_ways = 0;
+    let mut num_relations = 0;
+
+    for obj in pbf.par_iter() {
+        let obj = obj.unwrap();
+        match obj {
+            osmpbfreader::OsmObj::Node(_) => num_nodes += 1,
+            osmpbfreader::OsmObj::Way(_) => num_ways += 1,
+            osmpbfreader::OsmObj::Relation(_) => num_relations += 1,
+        }
+    }
+    assert_eq!(num_nodes, 3097);
+    assert_eq!(num_ways, 151);
+    assert_eq!(num_relations, 17);
+}
+
+#[test]
+fn read_pbf_highway() {
+    let file = File::open(PBF).unwrap();
+    let mut pbf = osmpbfreader::OsmPbfReader::new(file);
+
+    let mut num_nodes = 0;
+    let mut num_ways = 0;
+    let mut num_relations = 0;
+    let mut sum_ids = 0;
```

Comment:
Why not doing this sum_id check on the other tests ? If that’s an overflow problem, you can use `.wrapping_add(_)`.

Reply:
You didn’t do that.

Reply:
sum_node/way/relation_id are checked on all functions, thanks to the common Stats field.

I added the check on `sum_ids` only here, because `get_objs_and_deps()` is the only function providing both an OsmId and an OsmObj.